codecs/avif/enc/avif_enc.cpp

```diff
@@ -3,15 +3,31 @@
 #include <emscripten/val.h>
 #include "avif/avif.h"
 
+#include <string>
+
+#define RETURN_NULL_IF_NOT_EQUALS(val1, val2) \
+  do {                                        \
+    if (val1 != val2)                         \
+      return val::null();                     \
+  } while (false)
+#define RETURN_NULL_IF_EQUALS(val1, val2) \
+  do {                                    \
+    if (val1 == val2)                     \
+      return val::null();                 \
+  } while (false)
```

Comment:
Optional: You can merge these two macros into `RETURN_NULL_IF_FALSE` that takes a conditional expression as the argument.

Reply:
That's so much better than the current macro LOL, thank you!

Reply:
Note: I probably would not use a macro that saves only one line. But some people like to use this kind of macro so that the main flow of the code is not cluttered with error handling code.

Reply:
Yep that was my line of thinking. I'm pretty sure we'll be repeating that line again for other optimizations, so it's better to just have a macro for it.